File: src/main/java/com/shelterhub/dto/request/MedicalRecordRequest.java
package com.shelterhub.dto.request;

import com.shelterhub.domain.model.MedicalRecord;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MedicalRecordRequest {
    private UUID id;
    private UUID animalId;

    public MedicalRecord toMedicalRecord() {
        return MedicalRecord.builder()
                .id(getId())
                .animalId(getAnimalId())
                .build();
    }
}

File: src/main/kotlin/br/com/shelterhubmanagementapi/dto/request/MedicalRecordRequest.kt
package br.com.shelterhubmanagementapi.dto.request

import br.com.shelterhubmanagementapi.domain.model.MedicalRecord
import java.util.UUID

data class MedicalRecordRequest(
    val id: UUID = UUID.randomUUID(),
    val animalId: UUID
)

fun MedicalRecordRequest.toMedicalRecord() =
    MedicalRecord(
        id = this.id,
        animalId = this.animalId
    )
